### DISCUSSION TITLE
File loading Modified By throws an exception, because the User is not in AAD

### DISCUSSION THREAD

**Discussion by ChrTall** [pnp/pnpcore#780] [Q&A]
When I load File Properties of a folder I want to display all Items and Subfolders with their following Properties: Name, ModifiedBy, Size etc.
The problem I am facing is that we migrated Exisitng Data from a OnPremise Sharepoint to Sharepoint 365 and not every User that Modified a File is in AAD. When I view the Folder in Sharepoint Web UI there is a property with the name, but you can not click and expand the User like with Users that exist in AAD. 
I am using the following way to get the Files from the folder:
`var files = await currentFolder.Files
                .QueryProperties(file => file.Name, file => file.TimeLastModified, file => file.LinkingUrl, files => files.ServerRelativeUrl, file => file.Length)
                .QueryProperties(file => file.ModifiedBy)
                .ToListAsync();`

And then I iterate over them:
` await file.ModifiedBy.EnsurePropertiesAsync(modBy => modBy.Title);`

This throws a exception because the User is not in Azure Active Directory.
I don´t know if this is only a problem for old data that is migrated, because from time to time Users may get removed from AAD because they leave the company. Then you would also get this problem.

Is there another way to get the name of the user that modified the file?  It is metadata like for any other ListItem isn´t it?

**Comment by jansenbe** [comment]
@ChrTall : if the user does not exist anymore you cannot query it like that. Did you try loading the `IFile` `ListItemAllFields` property (the list item linked to the file) and see if you can get something useful from there?

**Reply by ChrTall** [reply]
@jansenbe I am currently catching the exeception when the user does not exist and then I use the IFIle.Properties Bag with the following key:
"display_urn_x003a_schemas_x002d_microsoft_x002d_com_x003a_office_x003a_office_x0023_Editor"   for File Last Modified By;

This pretty much works and solves my problem. Is this a safe way to rely on? The name of the key is a bit cryptic and I am worried, that it will breake in the future. Another problem with this approach I could yet not solve is that even though I use the Properties bag of IFIles only for the Files that throw the exception, I have to load the whole big Properties bag for every File, because 
`await file.EnsurePropertiesAsync(file => file.Properties);`
does not load the Properties bag if it was not loaded previously.
Also I was wondering if one can load only the specific key and entry instead of the whole Property Bag for the file.
Thank you for your help and patience.

**Reply by jansenbe** [reply]
@ChrTall : if the property is there on the all the places you need it then I would just use it. We'll not suddenly change it's name. About the second part: why don't you load the properties only when the exception occurs. Try using `await file.LoadAsync(file => file.Properties)`. At the moment we don't support loading a specific key from a property bag.
